mypy/literals.py

```diff
@@ -112,6 +113,8 @@ def literal(e: Expression) -> int:
             return LITERAL_NO
 
     elif isinstance(e, NameExpr):
+        if isinstance(e.node, Var) and e.node.is_final and e.node.final_value is not None:
```

Comment:
What other checks will be affected by this? 🤔 

Maybe we can add tests for them as well.

Reply:
Other cases would fallback to `LITERAL_TYPE`, which is much conservative than `LITERAL_YES`. I think this change only expose those `Final`s. 

As for this TypedDict case, `one_typeddict[general_var]` is forbidden since we only allow TypedDict key to be a string literal.

Reply:
I think this will affect e.g. `if foo[index] is not None` etc. where `index` is final for an arbitrary `foo` not just TypedDicts. You can add a test for a tuple with final integer position.

Reply:
Just added a test.